updater/dart_bindings/lib/src/bindings.dart

```diff
@@ -49,7 +52,13 @@ class AppParameters extends ffi.Struct {
     calloc.free(config.ref.base_version);
     calloc.free(config.ref.channel);
     calloc.free(config.ref.update_url);
-    calloc.free(config.ref.original_libapp_path);
+    // Free all paths in original_libapp_path.
+    var i = 0;
+    while (config.ref.original_libapp_paths[i].address != 0) {
```

Comment:
There are various ways to represent lists in C.  It looks like you're assuming that they're an array that's N+1 in length and the last entry is 0?